### DISCUSSION TITLE
When the devtoolextension installed?

### DISCUSSION THREAD

**Discussion by OriLiMu** [ScratchAddons/ScratchAddons#7564] [Help] [answered]
```js
  if (!addon.self._isDevtoolsExtension && window.initGUI) {
    console.log("Extension running, stopping addon");
    window._devtoolsAddonEnabled = true;
    window.dispatchEvent(new CustomEvent("scratchAddonsDevtoolsAddonStopped"));
    return;
  }
```

This code you can find here : https://github.com/ScratchAddons/ScratchAddons/blob/master/addons/find-bar/userscript.js#L5

About the definition of `_isDevtoolsExtension`, `initGUI`, and event `scratchAddonsDevtoolsAddonStopped` are related to the repo `DevtoolsExtension`. But when the devtool installed? 

I just drag the folder of `ScratchAddons` to chrome. And not found code in SA install the `DevtoolsExtension`.

**Comment by Samq64** [comment, marked as answer]
That can be removed at some point. The devtools addon used to be its own browser extension.
https://chromewebstore.google.com/detail/scratch-3-developer-tools/bfpdlicnalglgniemfjmbjbadekoilpl

**Reply by OriLiMu** [reply]
Do you know where is the definition  of `startBlock_`. It seems the property of `Blockly` which is a part of scratch code. 
But I search `startBlock_` in `scratch-gui` and not found it. 
Why?

**Reply by OriLiMu** [reply]
Sorry forget to paste the source code at first . 
https://github.com/ScratchAddons/ScratchAddons/blob/0f5b51e709a7f5b763db38fe6e9b562816292d99/addons/jump-to-def/userscript.js#L25

The definition of `startBlock_` is here. But I cant find where the `startBlock_` is assigned.

**Reply by CST1229** [reply]
> But I search `startBlock_` in `scratch-gui` and not found it.

Blockly is `scratch-blocks`, not `scratch-gui`.
